src/number.c

```diff
@@ -759,16 +762,22 @@ static char *number2Cstr(SCM n, long base, char buffer[], size_t bufflen)
       }
     case tc_complex:
       {
-        char *s1, *s2, *s3, tmp[100];
+        /* We print the realand imaginary parts in re and im, and then
+           glue them together in cplx. tmp is just a temporary buffer for
+           the imaginary part. */
+        char *re, *im, *cplx, tmp[100];
         size_t len;
 
-        s1  = number2Cstr(COMPLEX_REAL(n), base, buffer, bufflen);
-        s2  = number2Cstr(COMPLEX_IMAG(n), base, tmp, sizeof(tmp));
-        len  = strlen(s1) + strlen(s2) + 3;
-        s3 = STk_must_malloc_atomic(len);
-        snprintf(s3, len, "%s%s%si", s1, ((*s2 == '-') ? "": "+"), s2);
```

Comment:
This is clearer with `re` and `im,` but we lose the symmetry with what is done for rational. Perhaps, use `left`, `right`, `res` or both. What do you think @jpellegrini?

Reply:
Ah, yes. I thought it was kind of away from the style in the code. I'll redo that part.

Reply:
Is it OK now?  :)